From c21ee3827639343920d3f08cff319620d3582d2a Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Thu, 12 Nov 2020 16:19:36 -0800
Subject: ped_assert: Fix incorrect exception option

PED_EXCEPTION_FATAL is a type, not an option. A PED_EXCEPTION_BUG should
always select CANCEL.
---
 libparted/debug.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libparted/debug.c b/libparted/debug.c
index c4932ca..227673e 100644
--- a/libparted/debug.c
+++ b/libparted/debug.c
@@ -106,7 +106,7 @@ void ped_assert (const char* cond_text,
         /* Throw the exception */
         ped_exception_throw (
                 PED_EXCEPTION_BUG,
-                PED_EXCEPTION_FATAL,
+                PED_EXCEPTION_CANCEL,
                 _("Assertion (%s) at %s:%d in function %s() failed."),
                 cond_text, file, line, function);
         abort ();
-- 
cgit v1.2.1

